components/brave_wallet/browser/swap_service.cc

```diff
@@ -661,4 +666,40 @@ void SwapService::OnGetLiFiTransaction(GetTransactionCallback callback,
   }
 }
 
+void SwapService::GetLiFiStatus(const std::string& tx_hash,
+                                GetLiFiStatusCallback callback) {
+  auto conversion_callback = base::BindOnce(&ConvertAllNumbersToString, "");
+  auto internal_callback =
+      base::BindOnce(&SwapService::OnGetLiFiStatus,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  api_request_helper_.Request(net::HttpRequestHeaders::kGetMethod,
+                              GetLiFiStatusURL(tx_hash), "", "",
+                              std::move(internal_callback), GetHeaders(), {},
+                              std::move(conversion_callback));
+}
+
+void SwapService::OnGetLiFiStatus(GetLiFiStatusCallback callback,
+                                  APIRequestResult api_request_result) {
+  if (!api_request_result.Is2XXResponseCode()) {
+    if (auto error_response =
+            lifi::ParseErrorResponse(api_request_result.value_body())) {
+      std::move(callback).Run(nullptr, std::move(error_response), "");
+    } else {
+      std::move(callback).Run(
+          nullptr, nullptr, l10n_util::GetStringUTF8(IDS_WALLET_PARSING_ERROR));
+    }
+
+    return;
```

Comment:
nit: I'd suggest to pair `std::move(callback).Run(...)` with immediate `return`: it's a bit easier to review for missing returns and then possbile incorrect double runs of callbacks.

Though this file does not follow that.